From 7d0f337d75da9b3074b03e389a640b3745db2719 Mon Sep 17 00:00:00 2001
From: Daniel Drubin <idaniel.drubin@intel.com>
Date: Tue, 9 Feb 2016 18:41:49 +0200
Subject: [PATCH 07/12] ISH: fix HID client driver's init sequence

Fixed initialization order in HID client driver (initialize workqueue
before registering driver)

Change-Id: Ie71184547599c282f5aba0690cbaca65a1629e65
Signed-off-by: Daniel Drubin <daniel.drubin@intel.com>
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-19484
Reviewed-on: https://android.intel.com:443/468144
---
 drivers/misc/intel-ish/heci-hid-client.c | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/drivers/misc/intel-ish/heci-hid-client.c b/drivers/misc/intel-ish/heci-hid-client.c
index 3f4f97c..5f7ac44 100644
--- a/drivers/misc/intel-ish/heci-hid-client.c
+++ b/drivers/misc/intel-ish/heci-hid-client.c
@@ -786,16 +786,11 @@ static int __init ish_init(void)
 
 	init_waitqueue_head(&init_wait);
 	init_waitqueue_head(&heci_hid_wait);
+	INIT_WORK(&my_work, workqueue_init_function);
+
 	/* Register HECI client device driver - ISS */
 	rv = heci_cl_driver_register(&hid_heci_cl_driver);
 
-	/*
-	 * 7/7/2014: in order to not stick Android boot, from here & below
-	 * needs to run in work queue and here we should return rv
-	 */
-	/****************************************************************/
-	INIT_WORK(&my_work, workqueue_init_function);
-	/***************************************************************/
 	return rv;
 
 }
-- 
1.9.1

